plugins/pulp_rpm/plugins/migrations/0028_standard_storage_path.py

```diff
@@ -112,17 +94,70 @@ def yum_metadata_plan():
     return Plan(collection, key_fields)
 
 
-def iso_plan():
-    """
-    Factory to create an ISO migration plan.
-
-    :return: A configured plan.
-    :rtype: Plan
-    """
-    key_fields = (
-        'name',
-        'checksum',
-        'size'
-    )
-    collection = connection.get_collection('units_iso')
-    return Plan(collection, key_fields)
+class Distribution(Plan):
+    """
+    The migration plan for Distribution units.
+    """
+
+    def __init__(self):
+        """
+        Call super with collection and fields.
+        """
+        key_fields = (
+            'distribution_id',
+            'family',
+            'variant',
+            'version',
+            'arch'
+        )
+        collection = connection.get_collection('units_distribution')
+        super(Distribution, self).__init__(collection, key_fields, join_leaf=False)
+
+    def _new_path(self, unit):
+        """
+        The *variant* might not exist in the document for older units.
+        Default the variant part of the unit key to '' which matches the model.
+
+        :param unit: The unit being migrated.
+        :type  unit: pulp.plugins.migration.standard_storage_path.Unit
+        :return: The new path.
+        :rtype: str
+        """
+        unit.document.setdefault('variant', '')
+        return super(Distribution, self)._new_path(unit)
+
+
+class ISO(Plan):
+    """
+    The migration plan for ISO units.
+    """
+
+    def __init__(self):
+        """
+        Call super with collection and fields.
+        """
+        key_fields = (
+            'name',
+            'checksum',
+            'size'
+        )
+        collection = connection.get_collection('units_iso')
+        super(ISO, self).__init__(collection, key_fields)
+
+    def _new_path(self, unit):
+        """
+        Units created by 2.8.0 don't include the ISO name.  This was a regression
+        that is being corrected by this additional logic.  If the storage path
+        does not end with the *name* stored in the unit, it is appended.
+
+        :param unit: The unit being migrated.
+        :type  unit: pulp.plugins.migration.standard_storage_path.Unit
+        :return: The new path.
+        :rtype: str
+        """
+        name = unit.document['name']
+        path = unit.document['_storage_path']
+        if not path.endswith(name):
+            unit.document['_storage_path'] = name
```

Comment:
I see that the super class uses this when re-calculating the ``_storage_path``. It would be helpful to add a little comment here explaining what will happen, because otherwise this looks surprising and possibly wrong on first glance.